### PR TITLE
Add Feature plugin plumbing

### PR DESCRIPTION
**Pull request by psealock**
This PR gets the repo ready to be a feature plugin.

Note: We need linting and .editorconfig because its a bit all over the place. I'm placing "Needs review" label, but happy to wait until we get linting sorted before merging.

* Adds psr-4 
* Adds composer
* Creates FeaturePlugin and Loader classes similar to wc-admin

## Test

1. Activate this plugin.
2. Visit the plugins screen and see the message to run `composer install`.
3. `composer install`.
4. Reload the page and see "hello world" in the console.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{
+	"name": "woocommerce/navigation",
+	"description": "A modern, javascript-driven navigation experience.",
+	"homepage": "https://github.com/woocommerce/woocommerce-admin",
```
Comment (joshuatf) [inline composer.json:4]:
```suggestion
	"homepage": "https://github.com/woocommerce/navigation",
```

File: src/Loader.php [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	public static function get_instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Constructor.
+	 * Hooks added here should be removed in `wc_admin_initialize` via the feature plugin.
```
Comment (joshuatf) [inline src/Loader.php:35]: I think this comment can be removed.

File: woocommerce-navigation.php [hunk 3]
```diff
@@ -5,39 +5,53 @@
  * @package WC_Admin
```
Comment (joshuatf) [inline woocommerce-navigation.php:5]:
```suggestion
 * @package  WooCommerce Navigation
```

File: woocommerce-navigation.php [hunk 4]
```diff
@@ -5,39 +5,53 @@
...
-		return;
+$autoloader = __DIR__ . '/vendor/autoload.php';
+if ( is_readable( $autoloader ) ) {
+	require $autoloader;
+} else {
+	if ( defined( 'WP_DEBUG' ) && WP_DEBUG ) {
+		error_log(  // phpcs:ignore
+			sprintf(
+				/* translators: 1: composer command. 2: plugin directory */
+				esc_html__( 'Your installation of the WooCommerce Admin feature plugin is incomplete. Please run %1$s within the %2$s directory.', 'woocommerce-admin' ),
```
Comment (joshuatf) [inline woocommerce-navigation.php:26]:
```suggestion
				esc_html__( 'Your installation of the WooCommerce Navigation feature plugin is incomplete. Please run %1$s within the %2$s directory.', 'woocommerce-navigation' ),
```

File: src/Loader.php [hunk 5]
```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * Register the scripts, styles, and includes needed for pieces of the WooCommerce Navigation.
+ * NOTE: DO NOT edit this file in WooCommerce core, this is generated from woocommerce-admin.
+ *
+ * @package Woocommerce Admin
```
Comment (joshuatf) [inline src/Loader.php:6]:
```suggestion
 * @package Woocommerce Navigation
```

File: src/Loader.php [hunk 6]
```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * Register the scripts, styles, and includes needed for pieces of the WooCommerce Navigation.
+ * NOTE: DO NOT edit this file in WooCommerce core, this is generated from woocommerce-admin.
```
Comment (joshuatf) [inline src/Loader.php:4]:
```suggestion
 * NOTE: DO NOT edit this file in WooCommerce core, this is generated from woocommerce-navigation.
```

File: src/FeaturePlugin.php [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+		}
+		return static::$instance;
+    }
+    
+    /**
+	 * Init the feature plugin, only if we can detect both Gutenberg and WooCommerce.
+	 */
+	public function init() {
+        $this->define( 'WC_NAVIGATION_ABSPATH', dirname( __DIR__ ) . '/' );
+        $this->define( 'WC_NAVIGATION_PLUGIN_FILE', WC_NAVIGATION_ABSPATH . 'woocommerce-admin.php' );
```
Comment (joshuatf) [inline src/FeaturePlugin.php:51]:
```suggestion
        $this->define( 'WC_NAVIGATION_PLUGIN_FILE', WC_NAVIGATION_ABSPATH . 'woocommerce-navigation.php' );
```

File: woocommerce-navigation.php [hunk 8]
```diff
@@ -5,39 +5,53 @@
...
+	add_action(
+		'admin_notices',
+		function() {
+			?>
+			<div class="notice notice-error">
+				<p>
+					<?php
+					printf(
+						/* translators: 1: composer command. 2: plugin directory */
+						esc_html__( 'Your installation of the WooCommerce Navigation feature plugin is incomplete. Please run %1$s within the %2$s directory.', 'woocommerce-admin' ),
```
Comment (joshuatf) [inline woocommerce-navigation.php:44]:
```suggestion
						esc_html__( 'Your installation of the WooCommerce Navigation feature plugin is incomplete. Please run %1$s within the %2$s directory.', 'woocommerce-navigation' ),
```

**Comment by psealock** [comment]
>Happy to get this merged first to unblock us, but definitely agreed that we should that in place ASAP so we're not spending time reviewing formatting.

Fantastic, thanks for the review and suggestions. I quickly did this at end of day and didn't want to hold anyone up.

**Pull request merged**